From 3627d6d602e238e847573e404111421c1062ac24 Mon Sep 17 00:00:00 2001
From: Romain Perier <romain.perier@free-electrons.com>
Date: Fri, 26 Aug 2016 16:29:41 +0200
Subject: [PATCH 0612/1345] arm64: dts: marvell: add TRNG description for
 Armada 8K CP

This commits adds the devicetree description of the SafeXcel IP-76 TRNG
found in the two Armada CP110.

Change-Id: I56380832f424c25f350472fd250665a27fe246be
Signed-off-by: Romain Perier <romain.perier@free-electrons.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/32591
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi |   11 +++++++++++
 arch/arm64/boot/dts/marvell/armada-cp110.dtsi   |   11 +++++++++++
 2 files changed, 22 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
index 0cd2e1d..830b22a 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110-1.dtsi
@@ -313,6 +313,17 @@ ppv22@000000 {
 		status = "disabled";
 	};
 };
+
+cps_trng: trng@760000 {
+	compatible = "marvell,armada-8k-rng", "inside-secure,safexcel-eip76";
+	reg = <0x760000 0x7D>;
+	#address-cells = <1>;
+	#size-cells = <0>;
+	interrupts = <GIC_SPI 312 IRQ_TYPE_LEVEL_HIGH>;
+	clocks = <&cps_syscon0 1 25>;
+	status = "okay";
+};
+
 eip197@800000 {
 	compatible = "marvell,eip197";
 	reg = <0x800000 0x200000>;
diff --git a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
index d3a4a0f..24d0487 100644
--- a/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-cp110.dtsi
@@ -399,6 +399,17 @@ ppv22@000000 {
 		status = "disabled";
 	};
 };
+
+cpm_trng: trng@760000 {
+	compatible = "marvell,armada-8k-rng", "inside-secure,safexcel-eip76";
+	reg = <0x760000 0x7D>;
+	#address-cells = <1>;
+	#size-cells = <0>;
+	interrupts = <GIC_SPI 59 IRQ_TYPE_LEVEL_HIGH>;
+	clocks = <&cpm_syscon0 1 25>;
+	status = "okay";
+};
+
 eip197: eip197@800000 {
 	compatible = "marvell,eip197";
 	reg = <0x800000 0x200000>;
-- 
1.7.9.5

